### ISSUE TITLE
[Bug]: torchdynamo.export() doesn't work with torch specific constants 

### ISSUE THREAD

**Issue by tugsbayasgalan** [pytorch/pytorch#93564]
### 🐛 Describe the bug

```
def f(x):
    if torch.distributed.is_available():
        return x.cos() + 1
    return 1

torchdynamo.config.dynamic_shapes = True
torchdynamo.export(f, torch.randn(3, 4), aten_graph=True)
```

### Error logs
~~~
  File "/mnt/xarfuse/uid-23137/7756fc57-seed-nspid4026533012_cgpid17879168-ns-4026533009/torchdynamo/symbolic_convert.py", line 335, in run
    and self.step()
  File "/mnt/xarfuse/uid-23137/7756fc57-seed-nspid4026533012_cgpid17879168-ns-4026533009/torchdynamo/symbolic_convert.py", line 308, in step
    getattr(self, inst.opname)(inst)
  File "/mnt/xarfuse/uid-23137/7756fc57-seed-nspid4026533012_cgpid17879168-ns-4026533009/torchdynamo/symbolic_convert.py", line 149, in inner
    unimplemented(f"generic_jump {typestr(value)}")
  File "/mnt/xarfuse/uid-23137/7756fc57-seed-nspid4026533012_cgpid17879168-ns-4026533009/torchdynamo/exc.py", line 71, in unimplemented
    raise Unsupported(msg)
torchdynamo.exc.Unsupported: generic_jump DynamicShapeVariable()
~~~
### Did Dynamo succeed?

- [ ] Does dynamo.optimize("eager") succeed?

### Did AOT succeed?

- [ ] Did dynamo.optimize("aot_eager") succeed?

### Did Inductor succeed?

- [ ] Does dynamo.optimize("inductor") succeed?

### Minified repro

Repro above

cc @ezyang @soumith @msaroufim @wconstab @ngimel @bdhirsh

**Comment by ezyang** [comment]
works now

```
(/home/ezyang/local/a/pytorch-env) [ezyang@devgpu020.ftw1 ~/local/a/pytorch (f987a3be)]$ python l.py
cat l.py
(GraphModule(), {Guard(name='torch', source=<GuardSource.GLOBAL: 1>, create_fn=<function GuardBuilder.FUNCTION_MATCH at 0x7f3eda26c3a0>, is_volatile=False, guard_types=None, code_list=None, obj_weakref=None, guarded_class_weakref=None), Guard(name='x', source=<GuardSource.LOCAL: 0>, create_fn=<function GuardBuilder.TENSOR_MATCH at 0x7f3eda26ca60>, is_volatile=False, guard_types=['TENSOR_MATCH'], code_list=None, obj_weakref=<weakref at 0x7f3ed28889a0; to 'Tensor' at 0x7f3f6b558450>, guarded_class_weakref=<weakref at 0x7f3f30b14400; to 'torch._C._TensorMeta' at 0x3e44b50 (Tensor)>), Guard(name='', source=<GuardSource.SHAPE_ENV: 6>, create_fn=<function GuardBuilder.SHAPE_ENV at 0x7f3eda26c9d0>, is_volatile=False, guard_types=['SHAPE_ENV'], code_list=['x.stride()[0] == x.size()[1] and x.stride()[1] == 1 and x.storage_offset() == 0 and x.size()[0] != 0 and x.size()[0] != 1 and x.size()[1] != 0 and x.size()[1] != 1'], obj_weakref=None, guarded_class_weakref=None)})
(/home/ezyang/local/a/pytorch-env) [ezyang@devgpu020.ftw1 ~/local/a/pytorch (f987a3be)]$ cat l.py
import torch
import torch._dynamo.config

def f(x):
    if torch.distributed.is_available():
        return x.cos() + 1
    return 1

torch._dynamo.config.dynamic_shapes = True
print(torch._dynamo.export(f, torch.randn(3, 4), aten_graph=True))
```
